x/ibc/core/02-client/genesis.go

```diff
@@ -14,6 +14,12 @@ import (
 func InitGenesis(ctx sdk.Context, k keeper.Keeper, gs types.GenesisState) {
 	k.SetParams(ctx, gs.Params)
 
+	// Set all client metadata first. This will allow client keeper to overwrite client and consensus state keys
+	// if clients accidentally write to ClientKeeper reserved keys.
+	if len(gs.ClientsMetadata) != 0 {
```

Comment:
nit, this `if` is not strictly necessary

Reply:
True, but I think it helps for understandability so leaving as-is.